src/modules/sih/sih.cpp

```diff
@@ -83,6 +81,80 @@ void Sih::run()
 	_vehicle = (VehicleType)constrain(_sih_vtype.get(), static_cast<typeof _sih_vtype.get()>(0),
 					  static_cast<typeof _sih_vtype.get()>(2));
 
+
+#if defined(ENABLE_LOCKSTEP_SCHEDULER)
+	lockstep_loop();
+#else
+	realtime_loop();
+#endif
+}
+
+#if defined(ENABLE_LOCKSTEP_SCHEDULER)
+
+// Get current timestamp in microseconds
+uint64_t micros()
+{
+	struct timespec ts;
+	timespec_get(&ts, TIME_UTC);
+	return ((uint64_t)ts.tv_sec) * 1000000 + ((uint64_t)ts.tv_nsec) / 1000;
+}
+
+void Sih::lockstep_loop()
+{
+
+	int rate = math::min(_imu_gyro_ratemax.get(), _imu_integration_rate.get());
+
+	// default to 250Hz (4000 us interval)
+	if (rate <= 0) {
+		rate = 250;
```

Comment:
The hard coded update period in ekf2 is 10 milliseconds, so I would do a multiple (eg 400 Hz).